server/postgresql.go

```diff
@@ -0,0 +1,99 @@
+package server
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"gorm.io/datatypes"
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+)
+
+type Postgres struct {
+	db *gorm.DB
+}
+
+type PostgresConfig struct {
+	Host     string
+	Port     int
+	User     string
+	Password string
+	DBName   string
+}
+
+type DBTelemetry struct {
+	ID            int64          `gorm:"column:id; primaryKey"`
+	Application   string         `gorm:"column:application"`
+	TelemetryType string         `gorm:"column:telemetry_type"`
+	Time          time.Time      `gorm:"column:time"`
+	Payload       datatypes.JSON `gorm:"column:payload"`
+}
+
+func NewPostgres(config *PostgresConfig) StorageAdapter {
+	db, err := initDB(config)
+	if err != nil {
+		log.Fatalf("failed to connect database: %s", err)
+		return nil
+	}
+	err = migrateDB(db)
+	if err != nil {
+		log.Fatalf("failed to migrate database: %s", err)
+		return nil
+	}
+	return &Postgres{db: db}
+}
+
+func (p *Postgres) StoreHostTelemetry(h []*HostTelemetry) error {
+	if len(h) == 0 {
+		return nil
+	}
+	telemetryPayload, err := json.Marshal(h)
+	if err != nil {
+		return errors.Wrap(err, "failed to marshal telemetry payload")
+	}
+	sotrableTelemetry := &DBTelemetry{
+		Application:   "trento",
+		TelemetryType: "host_telemetry",
+		Time:          h[0].Time, // what is the time we want to track?
+		Payload:       telemetryPayload,
+	}
+	err = p.db.Create(sotrableTelemetry).Error
```

Comment:
We need to store a row for each `HostTelemetry` received, so basically iterating for each element in the slice and creating a slice of `DBTelemetry`

Reply:
Ah ok. Changed.